### DISCUSSION TITLE
How can I change the way % percent translated is calculated?

### DISCUSSION THREAD

**Discussion by working-name** [WeblateOrg/weblate#5432] [Q&A] [answered]
Hello there,

I have a simple JSON based project with English keys on the left, and the translation on the right for each language file. The right hand side copies the left initially, meaning `{ "english": "english" }` and that counts as translated for weblate. 

I'm trying to change it so that % translated only takes into account non-failing translations, and one of the hard fails is `string is the same`.

Any idea how I would be able to do that? 
Thank!

**Comment by nijel** [comment, marked as answer]
You might want to install addon that will mark such strings as needing editing, see https://docs.weblate.org/en/latest/admin/addons.html#flag-unchanged-translations-as-needs-editing

**Reply by working-name** [reply]
I'm sorry, no idea why but this is so not straightforward to me at all. Read the doc link, realized it's a settings.py thing to pick addons. I've also restarted uwsgi and celery and ran `pip install -U Weblate` just in case. I don't python, don't know its quirks. So:

1. Uncommented `"weblate.addons.flags.SameEditAddon"` for `WEBLATE_ADDONS`

```js
$ weblate list_addons
.. _addon-weblate.flags.same_edit:

Flag unchanged translations as "Needs editing"
----------------------------------------------

Whenever a new translatable string is imported from the VCS and it matches a
source string, it is flagged as needing editing in Weblate. This is especially
useful for file formats that include all strings even if not translated.
```

2. Added `strict-same` under Manage -> Settings -> Translation -> Translation flags of the component.
3. Reset repo which I assumed would trigger re-checking the strings. I was wrong, no change, translated is still at 100%.
4. Double checked the Manage -> Addons and it says `no addons currently installed` and `no compatible addons found`. Great, time to start over. Remove the component/project.

### Attempt no 2
1. Add project
2. Add component
3. Get into Manage -> Addons, still `There are no addons currently installed.` and `Could not find any additional addons compatible with this translation component.`
4. Add `strict-same` to Manage -> Settings -> Translation -> Translation flags
5. Reset repo

No change, weblate still thinks the project is 100% translated.

Why is the addons area behaving the way it is? What am I missing?

~~Can I consider that the addon is actually enabled since it's printed out by `weblate list_addons` and under Source Information box -> Flags there's `strict-same`?~~ No, it's not enabled.

```js
$ weblate install_addon --addon weblate.flags.same_edit site
Can not install on site/testaroo
```

**Update**: as a workaround in my case I could search for `has:check` and change status to `needs editing` via the bulk edit tool. Of course this won't be great if there's more than 1 check :)

**Reply by nijel** [reply]
> I'm sorry, no idea why but this is so not straightforward to me at all. Read the doc link, realized it's a settings.py thing to pick addons.

All built-in addons are automatically enabled, there is no need to edit configuration for that.

> Could not find any additional addons compatible with this translation component.

I see the addon is currently not allowed for bilingual translations, are you using that? (See https://docs.weblate.org/en/latest/formats.html#bimono). Anyway I've removed that limitation in 4d2cdb8cf9447ecd09f589c700848f2674fcd4d5.

**Reply by working-name** [reply]
> I see the addon is currently not allowed for bilingual translations, are you using that?

Yes, I am. The JSON format currently employed is

```JSON
{ "Hello!": "¡Hola!" }
```

Thank you very much for the help!
